### PR TITLE
Add level to API and CLI.

### PR DESCRIPTION
**Pull request by tdaede**
This validates MaxPicSize, MaxHSize, MaxVSize, and MaxDisplayRate.

MaxHeaderRate, Main/HighMbps, Main/HighCR, MaxTiles, and
MaxTileCols are not currently checked.

### CODE REVIEW

File: src/header.rs [hunk 1]
```diff
@@ -293,7 +293,7 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
       assert!(!fi.sequence.decoder_model_info_present_flag);
       assert_eq!(fi.sequence.operating_points_cnt_minus_1, 0);
       assert_eq!(fi.sequence.operating_point_idc[0], 0);
-      self.write(5, fi.sequence.level_idx[0])?; // level
+      self.write(5, 31)?; // level
```
Comment (lu-zero) [inline src/header.rs:296]: why is it 31?

File: src/api/config/mod.rs [hunk 2]
```diff
@@ -397,6 +407,36 @@ impl Config {
       }
     }
 
+    if let Some(level_idx) = config.level_idx {
+      if level_idx > 31 {
+        return Err(LevelUndefined);
+      }
+      if level_idx < 31 {
```
Comment (rzumer) [inline src/api/config/mod.rs:414]: Not sure if this is intentional for style, but that conditional is unnecessary.

File: src/bin/common.rs [hunk 3]
```diff
@@ -527,6 +532,24 @@ fn parse_config(matches: &CliOptions) -> Result<EncoderConfig, CliError> {
...
 
+  if let Some(level_str) = &matches.level {
+    cfg.level_idx = match &level_str.as_str() {
+      &"auto" => None,
+      &"unconstrained" => Some(31),
+      _ => {
+        let (major, minor) = scan_fmt!(&level_str, "{}.{}", u8, u8)
+          .expect("Could not parse AV1 level");
+        if major > 7 {
+          panic!("Invalid major AV1 level")
```
Comment (rzumer) [inline src/bin/common.rs:543]: Just my personal preference but I probably wouldn't bother specifying major or minor here, it's a bit arbitrary.

**Review by rzumer — APPROVED** [review]

File: src/api/config/mod.rs (hunk 2 above)
Comment (tdaede) [reply src/api/config/mod.rs:414]: it is ignoring the level_idx == 31 case

Comment (rzumer) [reply src/api/config/mod.rs:414]: Oh right my bad

File: src/header.rs (hunk 1 above)
Comment (tdaede) [reply src/header.rs:296]: avif used to always use level 31, guess it doesn't anymore

**Review by rzumer — APPROVED** [review]

**Pull request merged**